src/Agent.Plugins/PipelineCache/SavePipelineCacheV0.cs

```diff
@@ -29,6 +29,18 @@ public class SavePipelineCacheV0 : PipelineCacheTaskPluginBase
             string salt,
             CancellationToken token)
         {
+            if(!context.Variables.ContainsKey("agent.jobstatus"))
+            {
+                return;
+            }
+
+            var JobStatus = context.Variables["agent.jobstatus"];
+            if(JobStatus != null && !JobStatus.Value.Equals("Succeeded"))
```

Comment:
Feels like you could simplify all of this (lines 32-38) like this:

```cs
TaskResult? result = EnumUtil.TryParse<TaskResult>(context.variables[Constants.Variables.Agent.JobStatus]);
if (result == TaskResult.Succeeded) // or something close to this
{
   // cache
}
else
{
   // skip cache  
} 
```

Reply:
still need a context.Variables.ContainsKey(Constants.Variables.Agent.JobStatus) I think

---
In reply to: [299042822](https://github.com/microsoft/azure-pipelines-agent/pull/2320#discussion_r299042822) [](ancestors = 299042822)

Reply:
Yah yah :) Or like you said originally suggested ... something like `TryGetValue(Constants.Variables.Agent.JobStatus, out string jobStatus)`

Overall, seems like this can be simplified and we should be using constants/enum values instead of hard-coded strings.